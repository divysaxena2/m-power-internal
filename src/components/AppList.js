import React, { Component } from 'react';
import getTheme from '../../native-base-theme/components'
import material from '../../native-base-theme/variables/material';
import { StyleSheet, View } from 'react-native';
import { StyleProvider, List, ListItem, Left, Right, Body, Text } from 'native-base';


type Props = {};
export default class AppList extends Component<Props> {

    render() {
        return (
            <StyleProvider style={getTheme(material)}>
                <List>
                    <ListItem style={styles.listItem}>
                        <View style={{ paddingLeft: 15 }}>
                            <Text style={styles.date}>5 May,2019</Text>
                        </View>
                        <Body style={{ paddingLeft: 20 }}>
                            <Text style={styles.provider}>Kaiser Permanente</Text>
                            <Text note style={{ fontSize: 11 }}>Bill</Text>
                        </Body>
                        <View style={{ paddingLeft: 33 }}>
                            <Text style={styles.amount}>$500</Text>
                            <Text style={styles.payText}>Pay Now</Text>
                        </View>
                    </ListItem>
                </List>
            </StyleProvider>)
    }
}
const styles = StyleSheet.create({
    listItem: {
        height: 56,
        width: 363
    },
    date: {
        height: 16,
        width: 83.3,
        color: '#000000',
        fontSize: 12,
        letterSpacing: 0.47,
        lineHeight: 16
    },
    amount: {
        height: 20,
        width: 60,
        color: '#000000',
        //fontFamily: "Roboto Medium"; 
        fontSize: 15,
        letterSpacing: 1.06,
        lineHeight: 20,
        textAlign: 'right'

    },
    provider: {
        height: 16,
        width: 128,
        color: '#000000',
        fontSize: 12,
        letterSpacing: 0.47,
        lineHeight: 16
    },
    payText: {
        height: 15,
        opacity: 0.4,
        color: '#000000',
        // fontFamily: Roboto;
        fontSize: 11,
        letterSpacing: 1.02,
        lineHeight: 15,
        textAlign: 'right',
        textDecorationLine: 'underline'
    }
});