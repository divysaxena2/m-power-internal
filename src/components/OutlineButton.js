import React, { Component } from 'react';
import getTheme from '../../native-base-theme/components'
import material from '../../native-base-theme/variables/material';
import { StyleSheet, View } from 'react-native';
import { StyleProvider, Button, Text } from 'native-base';


type Props = {};
export default class OutlineButton extends Component<Props> {

    render() {
        return (
            <StyleProvider style={getTheme(material)}>
                <Button rounded light style={styles.buttonBody}>
                    <Text style={styles.buttonText}>Get Assistance</Text>
                </Button>
            </StyleProvider>)
    }
}
const styles = StyleSheet.create({
    buttonBody: {
        height: 36,
        width: 189,
        borderStyle: 'solid',
        borderColor: '#A6A5AC',
        borderWidth: 0.5,
        borderRadius: 18
    },
    buttonText: {
        height: 18,
        paddingLeft: 39,
        opacity: 0.54,
        color: '#000000',
        //font- family: "Roboto Light";
        fontSize: 13,
        letterSpacing: 0.62,
        lineHeight: 18,
        textAlign: 'center'
    }
});