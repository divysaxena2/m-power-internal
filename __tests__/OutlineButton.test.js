import 'react-native';
import React from 'react';
import AppList from '../src/components/AppList';

import renderer from 'react-test-renderer';
import OutlineButton from '../src/components/OutlineButton';

it('renders correctly', () => {
    const tree = renderer.create(
        <OutlineButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});