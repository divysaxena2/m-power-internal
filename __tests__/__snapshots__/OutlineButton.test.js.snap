// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f4f4f4",
      "borderBottomWidth": null,
      "borderColor": "#A6A5AC",
      "borderLeftWidth": null,
      "borderRadius": 18,
      "borderRightWidth": null,
      "borderStyle": "solid",
      "borderTopWidth": null,
      "borderWidth": 0.5,
      "elevation": 2,
      "flexDirection": "row",
      "height": 36,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.2,
      "width": 189,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#000",
          "fontFamily": "Roboto",
          "fontSize": 14,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "color": "#000000",
          "fontSize": 13,
          "height": 18,
          "letterSpacing": 0.62,
          "lineHeight": 18,
          "opacity": 0.54,
          "paddingLeft": 39,
          "textAlign": "center",
        },
      ]
    }
    uppercase={false}
  >
    Get Assistance
  </Text>
</View>
`;
